charts/kube-state-metrics/templates/_helpers.tpl

```diff
@@ -45,3 +45,17 @@ Allow the release namespace to be overridden for multi-namespace deployments in
     {{- .Release.Namespace -}}
   {{- end -}}
 {{- end -}}
+
+{{/* Generate basic labels */}}
+{{- define "kube-state-metrics.labels" }}
+app.kubernetes.io/name: {{ template "kube-state-metrics.name" . }}
+app.kubernetes.io/instance: "{{ .Release.Name }}"
+app.kubernetes.io/component: metrics 
+app.kubernetes.io/managed-by: "{{ .Release.Service }}"
+app.kubernetes.io/version: "{{ .Chart.AppVersion }}"
+app.kubernetes.io/part-of: {{ template "kube-state-metrics.name" . }} 
+helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+{{- if .Values.customLabels }}
+{{ toYaml .Values.customLabels }}
+{{- end }}
+{{- end }}
```

Comment:
The helm 3.6.0 default would make use of separated label vars which would look like:

```
{{/*
Common labels
*/}}
{{- define "kube-state-metrics.labels" -}}
helm.sh/chart: {{ include "kube-state-metrics.chart" . }}
{{ include "kube-state-metrics.selectorLabels" . }}
{{- if .Chart.AppVersion }}
app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
{{- end }}
app.kubernetes.io/managed-by: {{ .Release.Service }}
{{- end }}

{{/*
Selector labels
*/}}
{{- define "kube-state-metrics.selectorLabels" -}}
app.kubernetes.io/name: {{ include "kube-state-metrics.name" . }}
app.kubernetes.io/instance: {{ .Release.Name }}
{{- end }}
```

Reply:
@monotek i split the labels according Helm recommendation.